File: LeetCode/Easy/0844-backspace-string-compare/0844-backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        s_ = []
        t_ = []
        for i in range(len(s)):
            if s[i] == '#':
                if s_:
                    s_.pop()
            else:
                s_.append(s[i])

        for i in range(len(t)):
            if t[i] == '#':
                if t_:
                    t_.pop()
            else:
                t_.append(t[i])
        
        return s_ == t_

File: LeetCode/Easy/0844-backspace-string-compare/0844-backspace-string-compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        string s_;
        string t_;

        for(auto &str : s){
            if(str == '#') {
                if(!s_.empty()) {
                    s_.pop_back();
                }
            } else {
                s_.push_back(str);
            }
        }

        for(auto &str : t){
            if(str == '#') {
                if(!t_.empty()) {
                    t_.pop_back();
                }
            } else {
                t_.push_back(str);
            }
        }

        return s_ == t_;
    }
};